### PR TITLE
Feature/lu 4424 form status override

### PR DESCRIPTION
**Pull request by sivagudur**
Please review business layer api changes related to updating form status for override markers.

### CODE REVIEW

File: src/main/java/uk/gov/ons/collection/entity/ValidationOverride.java [hunk 1]
```diff
@@ -83,9 +91,18 @@ public ValidationOverride(String jsonString) throws InvalidJsonException {
                 }
             }
         }
+        log.info("Override Count {}", overrideCount);
         return updatedList;
     }
 
+    public String buildContributorStatusQuery(int triggerCount) {
+        log.info("Trigger Count {}", triggerCount);
+        String statusText = triggerCount == overrideCount ? STATUS_CLEAR_OVERRIDDEN : STATUS_CHECK_NEEDED ;
```
Comment (dllmorgan) [inline src/main/java/uk/gov/ons/collection/entity/ValidationOverride.java]: Might be worth considering putting some brackets around the logic check to make it a bit easier to identify on a scan-read.

File: src/main/java/uk/gov/ons/collection/service/ValidationOverrideService.java [hunk 2]
```diff
@@ -36,6 +36,16 @@ public void processValidationDataAndSave() throws Exception {
                     .qlSearch(updateQuery);
             log.info("Output after executing ValidationOutput UpdateGraphQl query " + validationOutputUpdateResponse);
         }
+
+        // Updating the Form Status for Override
+        int triggerCount = validationDatabaseList.size();
+        if (triggerCount > 0) {
```
Comment (dllmorgan) [inline src/main/java/uk/gov/ons/collection/service/ValidationOverrideService.java]: Does this comparison take into account the 'triggered=false' that are also in the outputs.

File: src/main/java/uk/gov/ons/collection/entity/ValidationOverride.java (hunk 1 above)
Comment (dllmorgan) [reply src/main/java/uk/gov/ons/collection/entity/ValidationOverride.java]: I think this calculation should be in its own method (with an input of trigger count) and outputting the String status and then this method should have an input of type String giving the status. Each method then has only 1 responsibility (and you can unit test the status derivation separately to building up the whole query).

Comment (sivagudur) [reply src/main/java/uk/gov/ons/collection/entity/ValidationOverride.java]: The above comments done.
1. Created a separate method 'processStatusMessage' with an input of trigger count.
2. Modified 'buildContributorStatusQuery' method with an input status text returned from the above method.
3. And also implemented few check style comments.

File: src/main/java/uk/gov/ons/collection/service/ValidationOverrideService.java (hunk 2 above)
Comment (sivagudur) [reply src/main/java/uk/gov/ons/collection/service/ValidationOverrideService.java]: The GraphQl query filters only triggered = true elements from database,  so validation outputs containing only trigger true elements.

**Pull request merged**